docs/command-reference/strings/getdel.md

```diff
@@ -8,28 +8,99 @@ import PageTitle from '@site/src/components/PageTitle';
+This atomically combines two operations — one to get the value of the key, and another to remove the key — making it more efficient than running the individual `GET` and `DEL` commands in sequence.
+
 ## Syntax
 
-    GETDEL key
+```shell
+GETDEL key
+```
+
+## Parameter Explanations
 
-**Time complexity:** O(1)
+- `key`: The key of the value you want to retrieve and delete.
 
-**ACL categories:** @write, @string, @fast
+## Return Values
 
-Get the value of `key` and delete the key.
-This command is similar to `GET`, except for the fact that it also deletes the key on success (if and only if the key's value type is a string).
+The command returns the value stored at the given key if the key exists.
+If the key does not exist, `nil` is returned.
+After the operation, the key is guaranteed to be deleted from the database.
 
-## Return
+## Code Examples
 
-[Bulk string reply](https://redis.io/docs/reference/protocol-spec/#bulk-strings): the value of `key`, `nil` when `key` does not exist, or an error if the key's value type isn't a string.
+### Basic Example
 
-## Examples
+Retrieve the value and delete the key:
 
 ```shell
-dragonfly> SET mykey "Hello"
+dragonfly> SET mykey "Hello, Dragonfly!"
 OK
+
 dragonfly> GETDEL mykey
-"Hello"
+"Hello, Dragonfly!"
+
 dragonfly> GET mykey
 (nil)
 ```
+
+In this example, the value `"Hello, Dragonfly!"` is returned by `GETDEL` and the `mykey` is deleted from the database, as verified by the subsequent `GET`.
+
+### Retrieving and Deleting a Non-Existent Key
+
+If the key does not exist, `GETDEL` will return `nil` without throwing an error:
+
+```shell
+dragonfly> GETDEL nonexistent_key
+(nil)
+```
+
+Since `nonexistent_key` does not exist, the return is `nil`, and no other action is taken.
+
+### Multiple Operations in Parallel
+
+If you operate on the same key with `GETDEL` and try to access it in parallel, the following result can occur:
+
+```shell
+dragonfly> SET anotherkey "Goodbye"
+OK
+
+dragonfly> GETDEL anotherkey
+"Goodbye"
+
+dragonfly> GET anotherkey
+(nil)
+```
+
+This ensures that both the retrieval and deletion occur atomically, confirming the key is no longer in the database afterward.
+
+## Best Practices
+
+- Use `GETDEL` when you want to both retrieve a value and delete the key in a single atomic operation, especially in scenarios where performing these two actions separately may introduce race conditions.
+- This operation is particularly helpful in cases like caching systems where you want to retrieve and invalidate cached values instantly.
+
+## Common Mistakes
+
+- Forgetting that after calling `GETDEL`, the key will no longer exist.
+  If subsequent operations depend on the key being available, you might need to handle it differently.
+- Assuming `GETDEL` will work on non-string data types like hashes, lists, or sets.
+  `GETDEL` only works on keys storing simple string values.
```

Comment:
Because "simple string" is a term in the RESP protocol. We also have things like "bulk string", even though they are all strings from a client SDK perspective. I would like to reduce possible confusion.